### ISSUE TITLE
Wrong sensor IDs for MetrLA?

### ISSUE THREAD

**Issue by radandreicristian** [liyaguang/DCRNN#70]
Are the sensors for MetrLA correct? In the file graph_sensor_ids.txt, all the sensor IDs start with 7, but it corresponds neither to the PeMS sensors nor the MetrLA sensors. Is there any use for that file?

**Comment by ThomasAFink** [comment]
Great question I also found they start with 7. Also are those the original speed timestamps in metr-la.h5 @liyaguang?

<img width="970" alt="Screenshot 2022-05-06 at 22 30 45" src="https://user-images.githubusercontent.com/53316058/167212374-174fbb3c-1eed-41c2-b705-091a34ec0c3c.png">


```
import pandas as pd
import h5py

#h5 file path
filename = 'metr-la.h5'

#read h5 file
dataset = h5py.File(filename, 'r')

#print the first unknown key in the h5 file
print(dataset.keys()) #returns df


#save the h5 file to csv using the first key df
with pd.HDFStore(filename, 'r') as d:
    df = d.get('df')
    df.to_csv('metr-la.csv')

```

**Comment by ThomasAFink** [comment]
The ids in LA distances don't start with 7. https://github.com/liyaguang/DCRNN/blob/master/data/sensor_graph/distances_la_2012.csv

**Comment by ThomasAFink** [comment]
> The ids in LA distances don't start with 7. https://github.com/liyaguang/DCRNN/blob/master/data/sensor_graph/distances_la_2012.csv

Never-mind they actually do further down the list.

**Comment by ThomasAFink** [comment]
> Great question I also found they start with 7. Also are those the original speed timestamps in metr-la.h5 @liyaguang?
> 
> <img alt="Screenshot 2022-05-06 at 22 30 45" width="970" src="https://user-images.githubusercontent.com/53316058/167212374-174fbb3c-1eed-41c2-b705-091a34ec0c3c.png">
> 
> ```
> import pandas as pd
> import h5py
> 
> #h5 file path
> filename = 'metr-la.h5'
> 
> #read h5 file
> dataset = h5py.File(filename, 'r')
> 
> #print the first unknown key in the h5 file
> print(dataset.keys()) #returns df
> 
> 
> #save the h5 file to csv using the first key df
> with pd.HDFStore(filename, 'r') as d:
>     df = d.get('df')
>     df.to_csv('metr-la.csv')
> ```

Okay it's speed data in 5min intervals: https://towardsdatascience.com/build-your-first-graph-neural-network-model-to-predict-traffic-speed-in-20-minutes-b593f8f838e5

**Comment by StefanBloemheuvel** [comment]
did anybody find a solution for this? Because now there are 4106 unique sensors in the dataset instead of the 207.

**Comment by ThomasAFink** [comment]
No, I got data from another city (Munich) and then built my own adjacency matrix: https://github.com/ThomasAFink/osmnx_adjacency_matrix_for_graph_convolutional_networks

**Comment by ThomasAFink** [comment]
Lots of data here from 40 cities: https://utd19.ethz.ch/

**Comment by radandreicristian** [comment]
@ThomasAFink @StefanBloemheuvel 
Sorry for the (very) late reply. I think I actually managed to get the data right somewhere last year. I actually have a repo where I have tried to get 2 of the datasets in an unified format with latest (back then) versions of NumPy. Everything I found prior to that had some version conflicts about how the data was generated/stored, so I did a remake.

Feel free to use/contribute:

https://github.com/radandreicristian/traffic-datasets
